tests/lib/prepare.sh

```diff
@@ -246,7 +249,7 @@ prepare_classic() {
         snapd_env="/etc/environment /etc/systemd/system/snapd.service.d /etc/systemd/system/snapd.socket.d"
         snap_confine_profiles="$(ls /etc/apparmor.d/snap.core.* || true)"
         # shellcheck disable=SC2086
-        tar cf "$SPREAD_PATH"/snapd-state.tar.gz /var/lib/snapd "$SNAP_MOUNT_DIR" /etc/systemd/system/"$escaped_snap_mount_dir"-*core*.mount /etc/systemd/system/multi-user.target.wants/"$escaped_snap_mount_dir"-*core*.mount $snap_confine_profiles $snapd_env
+        tar cf "$SPREAD_PATH"/snapd-state.tar /var/lib/snapd "$SNAP_MOUNT_DIR" /etc/systemd/system/"$escaped_snap_mount_dir"-*core*.mount /etc/systemd/system/multi-user.target.wants/"$escaped_snap_mount_dir"-*core*.mount $snap_confine_profiles $snapd_env
```

Comment:
I'm not familiar with the internals of our prepare/restore scripts so please forgive me the naive question: why do we have a mix of .tar-based snapshot and a simple recursive copy from save_all_snap_state and restore_all_snap_state and not just settle on one method?